p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/transport/RemoteArtifactRepositoryManagerAgentFactory.java

```diff
@@ -45,14 +49,33 @@ public Object createService(IProvisioningAgent agent) {
 	}
 
 	private boolean getDisableP2MirrorsConfiguration() {
-		String key = "tycho.disableP2Mirrors";
-		String value = System.getProperty(key);
-		if (value != null) {
-			logger.info("Using " + key
+		String deprecatedKey = "tycho.disableP2Mirrors";
+		String deprecatedValue = System.getProperty(deprecatedKey);
+
+		if (deprecatedValue == null) {
+			deprecatedValue = mavenContext.getSessionProperties().getProperty(deprecatedKey);
+		}
+
+		if (deprecatedValue != null) {
+			logger.info("Using " + deprecatedKey
 					+ " to disable P2 mirrors is deprecated, use the property eclipse.p2.mirrors instead, see https://tycho.eclipseprojects.io/doc/"
 					+ TychoVersion.getTychoVersion() + "/SystemProperties.html for details.");
-			return Boolean.parseBoolean(value);
+			return Boolean.parseBoolean(deprecatedValue);
+		}
+
+		String key = "eclipse.p2.mirrors ";
+		String value = System.getProperty(key);
+
+		if (value == null) {
+			value = mavenContext.getSessionProperties().getProperty(key);
+		}
+
+
+		if (value != null) {
+			// eclipse.p2.mirrors false -> disable mirrors
```

Comment:
One problem is that this will not really fully disable P2 mirrors so we probably need a P2 change here as well to make it really configurable.

Reply:
Not sure what to do here.

Reply:
if you search p2 code for `eclipse.p2.mirrors` you will see that this is only evaluated once as a system property, so even if you specify it Tycho will ignore mirrors, but P2 will still try to fetch the mirror list.